### PR TITLE
Bug/fix center camera

### PR DESCRIPTION
**Pull request by AlbertVVila**
:D

### CODE REVIEW

File: Source/ComponentCamera.cpp [hunk 1]
```diff
@@ -107,46 +107,50 @@ void ComponentCamera::Zoom(float mouseWheel)
...
+			childBboxes.Enclose(child->GetBoundingBox());
+		}
+		if (childBboxes.Volume() > 0)
+		{
+			CenterBbox(childBboxes);
+		}
+		else
+		{
+			float camDist = App->renderer->current_scale;
+			float3 center = ((ComponentTransform*)(App->scene->selected->GetComponent(ComponentType::Transform)))->position;
```
Comment (AlfonsoMestres) [inline Source/ComponentCamera.cpp]: Line size not in [Coding style](https://github.com/fractal-puppy/Engine/wiki/Cpp-style-guide#line-length)

File: Source/ComponentCamera.h [hunk 2]
```diff
@@ -35,7 +35,9 @@ class ComponentCamera :
 	math::float4x4 GetViewMatrix() const;
 	math::float4x4 GetProjectionMatrix() const;
 	math::LineSegment DrawRay(float x, float y) const;
+
 private:
+	void CenterBbox(const math::AABB & bbox);
```
Comment (AlfonsoMestres) [inline Source/ComponentCamera.h]: The & should be [closer to variable type](https://github.com/fractal-puppy/Engine/wiki/Cpp-style-guide#local-variables) and not in the middle of both.

File: Source/ComponentCamera.cpp [hunk 3]
```diff
@@ -107,46 +107,50 @@ void ComponentCamera::Zoom(float mouseWheel)
...
+		for (const auto &child : App->scene->selected->children)
 		{
-			GameObject* currentGo = descendents.top();
-			descendents.pop();
-			childBboxes.Enclose(currentGo->GetBoundingBox());
-			for (const auto &child : currentGo->children)
-			{
-				descendents.push(child);
-			}
+			childBboxes.Enclose(child->GetBoundingBox());
```
[the remark this answers is not in the document]
Comment (AlfonsoMestres) [reply Source/ComponentCamera.cpp]: This function is not being used for rendering purposes. We are managing the childBoxes in order to focus the camera when the object that want to be focused does not contains any mesh. 
The order is:
1- The object contain mesh? 
-> YES - Focus it
-> NO - Does it contains child meshes? 
--------> NO - Focus his transform
--------> YES - Gather all the children meshes and pick the average center.

[the remark this answers is not in the document]
Comment (AlbertVVila) [reply Source/ComponentCamera.cpp]: 10/10 couldn't explain it better

File: Source/ComponentCamera.cpp [hunk 4]
```diff
@@ -107,46 +107,50 @@ void ComponentCamera::Zoom(float mouseWheel)
...
-		frustum->pos = center + float3(0.0f, 0.0f, camDist);
 	}
 
 	frustum->front = -float3::unitZ;
 	frustum->up = float3::unitY;
 }
 
+void ComponentCamera::CenterBbox(const math::AABB& bbox)
+{
+	float3 HalfSize = bbox.HalfSize();
```
[the remark this answers is not in the document]
Comment (AlbertVVila) [reply Source/ComponentCamera.cpp]: 😭

File: Source/ComponentCamera.cpp [hunk 5]
```diff
@@ -107,46 +107,50 @@ void ComponentCamera::Zoom(float mouseWheel)
...
+	float distX = HalfSize.x / tanf(frustum->horizontalFov*0.5f);
+	float distY = HalfSize.y / tanf(frustum->verticalFov*0.5f);
+	float camDist = MAX(distX, distY) + HalfSize.z; //camera distance from model
+
+	float3 center = bbox.FaceCenterPoint(5);
```
[the remark this answers is not in the document]
Comment (AlbertVVila) [reply Source/ComponentCamera.cpp]: 😢

File: Source/ComponentCamera.cpp (hunk 4 above)
[the remark this answers is not in the document]
Comment (AlbertVVila) [reply Source/ComponentCamera.cpp]: Done!

File: Source/ComponentCamera.cpp (hunk 1 above)
Comment (AlbertVVila) [reply Source/ComponentCamera.cpp]: Done!

**Pull request merged**